services/kaze/controllers/application.go

```diff
@@ -228,29 +230,36 @@ func TransferApplicationOwnership(c *gin.Context) {
 
 // FetchMetrics retrieves the metrics of an application's container
 func FetchMetrics(c *gin.Context) {
+	c.Writer.Header().Set("Content-Type", "text/event-stream")
 	appName := c.Param("app")
-	filter := utils.QueryToFilter(c.Request.URL.Query())
-	var timeSpan int64
-
-	for unit, converter := range timeConversionMap {
-		if val, ok := filter[unit].(string); ok {
-			timeVal, err := strconv.ParseInt(val, 10, 64)
-			if err != nil {
-				continue
-			}
-			timeSpan += timeVal * converter
-		}
+
+	metricsFetchInterval := c.Query("fetch_interval")
+	fetchInt, err := strconv.ParseInt(metricsFetchInterval, 10, 64)
+	if err != nil {
+		fetchInt = 2 * int64(configs.ServiceConfig.Mizu.MetricsInterval)
 	}
+	fetchIntTime := time.Duration(fetchInt)
 
-	metrics := mongo.FetchContainerMetrics(types.M{
-		mongo.NameKey: appName,
-		mongo.TimestampKey: types.M{
-			"$gte": time.Now().Unix() - timeSpan,
-		},
-	})
+	metricsCount := c.Query("fetch_count")
+	count, err := strconv.ParseInt(metricsCount, 10, 64)
+	if err != nil {
+		count = 10
+	}
 
-	c.JSON(200, gin.H{
-		"success": true,
-		"data":    metrics,
+	c.Stream(func(w io.Writer) bool {
+		metrics := mongo.FetchContainerMetrics(types.M{
+			mongo.NameKey: appName,
+			mongo.TimestampKey: types.M{
+				"$gte": time.Now().Unix() - int64(configs.ServiceConfig.Mizu.MetricsInterval*time.Second),
+			},
+		}, count)
+		c.SSEvent("metrics", metrics)
+
+		if fetchIntTime < configs.ServiceConfig.Mizu.MetricsInterval {
+			fetchIntTime = 2 * fetchIntTime
+		}
+		time.Sleep(configs.ServiceConfig.Mizu.MetricsInterval * time.Second * fetchIntTime)
```

Comment:
Is it necessary to multiply `configs.ServiceConfig.Mizu.MetricsInterval`  and `fetchIntTime ` in this context ?